src/webgldrawer.js

```diff
@@ -876,101 +871,83 @@
-                }
-                return;
-            }
-
-            let textureInfo = this._TextureMap.get(canvas);
-
-            // if this is a new image for us, create a texture
-            if(!textureInfo){
-                let gl = this._gl;
+        _setupTextureHandlers() {
+            const _this = this;
+            const tex2DCompatibleLoader = (tile, data) => {
+                let tiledImage = tile.tiledImage;
+                let gl = _this._gl;
 
                 // create a gl Texture for this tile and bind the canvas with the image data
                 let texture = gl.createTexture();
                 let position;
                 let overlap = tiledImage.source.tileOverlap;
-
-                // deal with tiles where there is padding, i.e. the pixel data doesn't take up the entire provided canvas
-                let sourceWidthFraction, sourceHeightFraction;
-                if (tile.sourceBounds) {
-                    sourceWidthFraction = Math.min(tile.sourceBounds.width, canvas.width) / canvas.width;
-                    sourceHeightFraction = Math.min(tile.sourceBounds.height, canvas.height) / canvas.height;
-                } else {
-                    sourceWidthFraction = 1;
-                    sourceHeightFraction = 1;
-                }
-
                 if( overlap > 0){
                     // calculate the normalized position of the rect to actually draw
                     // discarding overlap.
-                    let overlapFraction = this._calculateOverlapFraction(tile, tiledImage);
-
-                    let left = (tile.x === 0 ? 0 : overlapFraction.x) * sourceWidthFraction;
-                    let top = (tile.y === 0 ? 0 : overlapFraction.y) * sourceHeightFraction;
-                    let right = (tile.isRightMost ? 1 : 1 - overlapFraction.x) * sourceWidthFraction;
-                    let bottom = (tile.isBottomMost ? 1 : 1 - overlapFraction.y) * sourceHeightFraction;
-                    position = this._makeQuadVertexBuffer(left, right, top, bottom);
-                } else if (sourceWidthFraction === 1 && sourceHeightFraction === 1) {
-                    // no overlap and no padding: this texture can use the unit quad as its position data
-                    position = this._unitQuad;
+                    let overlapFraction = _this._calculateOverlapFraction(tile, tiledImage);
+
+                    let left = tile.x === 0 ? 0 : overlapFraction.x;
+                    let top = tile.y === 0 ? 0 : overlapFraction.y;
+                    let right = tile.isRightMost ? 1 : 1 - overlapFraction.x;
+                    let bottom = tile.isBottomMost ? 1 : 1 - overlapFraction.y;
+                    position = _this._makeQuadVertexBuffer(left, right, top, bottom);
                 } else {
-                    position = this._makeQuadVertexBuffer(0, sourceWidthFraction, 0, sourceHeightFraction);
```

Comment:
It might've been lost in some of the pull requests I guess? Good catch, thanks. It is not easy to keep up with some changes, considering how far this PR reaches across the codebase.